util/settings/settings.go

```diff
@@ -1282,13 +1282,16 @@ func ReplaceStringSecret(val string, secretValues map[string]string) string {
 	if val == "" || !strings.HasPrefix(val, "$") {
 		return val
 	}
+	if !strings.HasPrefix(val, "$") {
+		return strings.TrimSpace(val)
+	}
```

Comment:
I think this code will actually never be executed, because the condition `!strings.HasPrefix(val, "$")` is already tested for in the `if` branch above.